File: pet_hospital_backend/src/main/java/com/pethospital/entity/Notification.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("notifications")
public class Notification {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long customerId;
    
    private String type; // APPOINTMENT, REMINDER, CONSULTATION, SYSTEM
    
    private String title;
    
    private String content;
    
    private String relatedType;
    
    private Long relatedId;
    
    private Integer isRead; // 1=已读, 0=未读
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime readAt;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: pet-hospital-miniprogram/src/api/notification.ts
import request from './request'

export interface Notification {
    id?: number
    customerId?: number
    type: string
    title: string
    content: string
    relatedType?: string
    relatedId?: number
    isRead?: number
    readAt?: string
    createdAt?: string
}

export const notificationApi = {
    // 获取我的通知列表
    getMyNotifications: (params?: { page?: number; size?: number }) => {
        return request.get('/api/notifications/my-notifications', params)
    },

    // 标记已读
    markAsRead: (id: number) => {
        return request.put(`/api/notifications/${id}/read`)
    },

    // 全部标记已读
    markAllAsRead: () => {
        return request.put('/api/notifications/read-all')
    },

    // 获取未读数量
    getUnreadCount: () => {
        return request.get('/api/notifications/unread-count')
    }
}
